unitex/src/fr/umlv/unitex/tfst/tagging/TaggingModel.java

```diff
@@ -105,27 +150,677 @@ protected void updateModel() {
+			== gb.getBounds().getStart_in_tokens() + tokensList.size() - 1) {
+			
+		    for( String s : tokensList ) {
+	            	if( !tokens.containsKey(index) )  
+	            		tokens.put(index, new ArrayList<String>());
+	            	if( !tokens.get(index).contains(s))
+	            		tokens.get(index).add(s);
+	            	index++;
+	            }
+			} else
+			if( gb.getBounds().getEnd_in_tokens()
+			== gb.getBounds().getStart_in_tokens() ){
+				if( !tokens.containsKey(index) )  
+            		tokens.put(index, new ArrayList<String>());
+            	if( !tokens.get(index).contains(temp))
+            		tokens.get(index).add(temp);
+			} else {
+				// implicit case where the difference is 0 < D < tokensList.length
+				tokens.put(index, new ArrayList<String>());
+				tokens.get(index).add(temp);
+			}
+			
+		}
+		for ( int i=0; i<tokens.size();i++ )
+	    	if( !tokens.containsKey(i) ) {
+	    		tokens.put( i, new ArrayList<String>());
+	    		tokens.get(i).add(" ");
+	    }
+	}
+	
+	/**  This function creates all the paths we can have between the first box
+	 *  "bfp" and the last box "bfs" and put them in allPaths. Then, it keeps only
+	 *  paths which contains at least a box which is tagged USELESS
+	 * 
+	 * @param bfp 
+	 * 			The number of the first box in the path
+	 * 
+	 * @param bfs
+	 * 			The number of the last box in the path
+	 * 
+	 * @param allPaths
+	 * 			The list of all path which contains at least a box USELESS
+	 */		
+	void computeAllPaths( int bfp, int bfs, ArrayList<ArrayList<Integer>> allPaths ){
+		ArrayList<ArrayList<Integer>> lstRemove = new ArrayList<>();
+		
+		computePath(boxes[sortedNodes[bfp]], boxes[sortedNodes[bfs]], new ArrayList<Integer>(), allPaths);
+ 		for (int tmp = 0; tmp < allPaths.size(); tmp++) {
+			allPaths.get(tmp).remove(0);
+			if(!containsUseless(allPaths.get(tmp)))
+				lstRemove.add(allPaths.get(tmp));
+		}
+		for(int i = 0; i < lstRemove.size(); i++)
+			allPaths.remove(lstRemove.get(i));
+		
+		for (ArrayList<Integer> arrayList : allPaths) {
+			for (int i = 0; i < arrayList.size(); i++) {
+				if(taggingStates[sortedNodes[arrayList.get(i)]] != TaggingState.NEUTRAL
+						&& taggingStates[sortedNodes[arrayList.get(i)]] != TaggingState.PREFERRED)
+					setBoxStateInternal(sortedNodes[arrayList.get(i)], TaggingState.TO_CHECK);
+			}
+			
+		}
+	}
+	
+	/** @Aissa
+	 * 
+	 * This function checks if the at least one box of the boxes in arrayList
+	 * is in state "USELESS"
+	 * 
+	 * @param arrayList 
+	 * 			ArrayList of the number of each boxes of the path
+	 * 
+	 * @return  true if at least one box is USELESS
+	 * 			false if not
+	 */			
+	private boolean containsUseless(ArrayList<Integer> arrayList) {
+		for(int i = 0; i < arrayList.size(); i++) {
+			if(taggingStates[sortedNodes[arrayList.get(i)]] == TaggingState.USELESS)
+				return true;
+		}
+		return false;
+	}
+
+	/** This function creates all paths between the first box bfp and the last box bfs 
```

Comment:
rename first two parameters as `fb1` and `fb2` for instance, and mention they are factorization nodes